src/announce_entry.cpp

```diff
@@ -91,8 +91,7 @@ namespace libtorrent {
 	void announce_endpoint::reset()
 	{
 		start_sent = false;
-		next_announce = time_point32::min();
-		min_announce = time_point32::min();
+		min_announce = next_announce = time_point32::min();
```

Comment:
just a comment, my personal preference is the more explicit nature of per field value set

Reply:
next_announce = time_point32::min();
min_announce = next_announce; 
?

Reply:
I rather do nothing, but is this a problem of performance? or just not to repeat the expression?

Reply:
https://www.viva64.com/en/w/v656/print/

V656 Variables 'next_announce', 'min_announce' are initialized through the call to the same function. It's probably an error or un-optimized code. Consider inspecting the 'time_point32::min()' expression. Check lines: 94, 95. announce_entry.cpp 95

![default](https://user-images.githubusercontent.com/920705/35169306-204b8114-fd6d-11e7-97ae-2a6d44ba1dfe.png)

Reply:
well...it's not my call :)

Reply:
``time_point32::min()`` is a constant (``constexpr`` function). There's no call. The diagnostic suggesting it's "unoptimized" is just not sophisticated enough. As for it being a potential mistake, I can see why that might justify the warning, but it isn't a mistake in this case.